### ISSUE TITLE
Add Sarus support?

### ISSUE THREAD

**Issue by marcodelapierre** [singularityhub/singularity-hpc#381]
Sarus is the container HPC runtime developed at the Swiss Supercomputing Centre, CSCS: https://sarus.readthedocs.io/en/stable/

**Comment by marcodelapierre** [comment]
just a note, I think I will be able to help with testing for this one, I can get a Sarus deployment on a VM if needed.

**Comment by vsoch** [comment]
Great! I've never used sarus before, but I'm planning on installing it on my local machine. Which (by the end of this project) will be chock full of different container technologies!

**Comment by vsoch** [comment]
Note to link to issue - sarus does not currently support pulling a specific digest: https://github.com/eth-cscs/sarus/issues/1. We can try pulling tags for now (and putting a warning) but hopefully this will be supported at some point!

**Comment by marcodelapierre** [comment]
Starting to document functionalities in Sarus, by memory of ~ 2 yo tests I ran, 
and also using a bit of docos: https://sarus.readthedocs.io/en/stable/user/user_guide.html   :

* pull: `sarus pull <repo>:<tag>` (digests still not supported apparently)
* execute: `sarus run <repo>:<tag> <cmd>`
* shell: `sarus run -t <repo>:<tag>  <shell>`
* execute default command:`sarus run <repo>:<tag>` (OCI behaviour as Docker; entrypoint can be overridden with `--entrypoint=<cmd>`, and removed with `--entrypoint ""`)
* add variables in container: apparently N/A
* default work directory: can be forced using `-w <directory>`
* host directory mounts: can be set with syntax like `--mount=type=bind,source=/path/to/my/data,destination=/data`
* GPU apps: work out of the box if Shifter is configured for that

**Comment by vsoch** [comment]
Yeah the non pull of a digest was unfortunately my stopping point. Although to think of it now, we could probably try something similar to Podman with pulling and then tagging.
